widgetssdk/src/main/java/com/glia/widgets/core/screensharing/data/GliaScreenSharingRepository.java

```diff
@@ -80,7 +80,11 @@ public void onScreenSharingAcceptedAndPermissionAsked(
 
     public void onScreenSharingDeclined() {
         Logger.d(TAG, "screen sharing declined by the user");
-
+        gliaCore.getCurrentEngagement().ifPresent(engagement -> {
+            // Pass RESULT_CANCELED to Core SDK to stop waiting for permission result. Otherwise, subsequent screen sharing requests won't be shown to the visitor.
```

Comment:
Thank you @andrews-moc for a very simple but clear explanation suggestion. I gust added the bug ticket for extra reference